components/script/dom/mediasession.rs

```diff
@@ -194,6 +196,76 @@ impl MediaSessionMethods for MediaSession {
             None => self.action_handlers.borrow_mut().remove(&action.into()),
         };
     }
+
+    /// https://w3c.github.io/mediasession/#dom-mediasession-setpositionstate
+    fn SetPositionState(&self, state: &MediaPositionState) -> Fallible<()> {
+        let duration: f64 = state.duration.map(|d| *d).unwrap_or(0.0);
+        let mut position: f64 = state.position.map(|p| *p).unwrap_or(0.0);
+        let mut playback_rate: f64 = state.playbackRate.map(|pr| *pr).unwrap_or(0.0);
+
+        // If the state is an empty dictionary then clear the position state.
+        if state.duration.is_none() && state.position.is_none() && state.playbackRate.is_none() {
```

Comment:
https://w3c.github.io/mediasession/#dom-mediasession-setpositionstate

> If the state is an empty dictionary then clear the position state.

Is the position state the combination of duration and playback_position and playbackRate in media_instance(type is HTMLMediaElement)?
And, Does clear mean to set at 0?

Reply:
Thanks! I added the reset method, and fixed to use it.